orttraining/orttraining/core/graph/optimizer_builder.cc

```diff
@@ -19,10 +19,11 @@ void OptimizerBuilderRegistry::RegisterBuilders() {
 Status IsMatchingTypeAndShape(
     const onnxruntime::Tensor& tensor,
     const int32_t element_type,
-    const std::vector<int64_t>& expected_shape) {
+    const std::vector<int64_t>& expected_shape_vec) {
   ORT_RETURN_IF_NOT(tensor.GetElementType() == element_type);
-  const std::vector<int64_t>& tensor_shape = tensor.Shape().GetDims();
-  ORT_RETURN_IF_NOT(tensor_shape == expected_shape);          
+  const TensorShape& tensor_shape = tensor.Shape();
+  const TensorShape& expected_shape(expected_shape_vec);
```

Comment:
>const TensorShape& expected_shape [](start = 2, length = 33)

if constructing a TensorShape here, this variable shouldn't be a reference (though it works for const references, [https://abseil.io/tips/107](https://abseil.io/tips/107)).
However there is also TensorShape::ReinterpretBaseType() for which a const reference is appropriate.